rocAL/source/augmentations/audio_augmentations/node_spectrogram.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <vx_ext_rpp.h>
+#include "node_spectrogram.h"
+#include "exception.h"
+
+SpectrogramNode::SpectrogramNode(const std::vector<Tensor *> &inputs, const std::vector<Tensor *> &outputs) :
+    Node(inputs, outputs) { }
+
+void SpectrogramNode::create_node() {
+    if(_node)
+        return;
+
+    vx_status status = VX_SUCCESS;
+    _window_fn_vx_array = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_FLOAT32, _window_length);
+    status |= vxAddArrayItems(_window_fn_vx_array, _window_length, _window_fn.data(), sizeof(vx_float32));
+    if(status != 0)
+        THROW(" vxAddArrayItems failed in the spectrogram node (vxRppSpectrogram)  node: "+ TOSTR(status) + "  " + TOSTR(status))
+    vx_scalar center_windows = vxCreateScalar(vxGetContext((vx_reference)_graph->get()), VX_TYPE_BOOL, &_is_center_windows);
+    vx_scalar reflect_padding = vxCreateScalar(vxGetContext((vx_reference)_graph->get()), VX_TYPE_BOOL, &_is_reflect_padding);
+    vx_scalar spectrogram_layout = vxCreateScalar(vxGetContext((vx_reference)_graph->get()), VX_TYPE_UINT32, &_spectrogram_layout);
```

Comment:
Enums are int, any specific reason we have the type as UINT32 for the layout of spectrogram?